### PR TITLE
Fix namespace pollution

### PR DESCRIPTION
**Pull request by arporter**
This is for #3 and adds the GO_ prefix to named constants to avoid clashing with Fortran reserved keywords.

### CODE REVIEW

File: finite_difference/src/kernel_mod.f90 [hunk 1]
```diff
@@ -31,12 +31,13 @@ module kernel_mod
 integer, public, parameter :: INTERNAL_PTS = 0, &
```
Comment (hiker) [inline finite_difference/src/kernel_mod.f90]: Shouldn't POINTWISE, DOFS, etc all have a GO_ as well?

File: finite_difference/src/global_parameters_mod.f90 [hunk 2]
```diff
@@ -16,17 +16,17 @@ module global_parameters_mod
   integer, parameter, public :: ALIGNMENT = 4
```
Comment (hiker) [inline finite_difference/src/global_parameters_mod.f90]: Shouldn't ALIGNMENT also have a GO_? Also, NAME_LEN appears to be unused, is this needed as part of this library?

**Review by hiker — COMMENTED** [review]
Great to get rid of write, min etc. A few questions - I somehow feel that you have not changed some of the names on purpose, but I thought I'd better ask :)

File: finite_difference/src/kernel_mod.f90 (hunk 1 above)
Comment (arporter) [reply finite_difference/src/kernel_mod.f90]: Yes, I'm in two minds about this (you'll see my last commit changed this for instance). I was going to do it for everything but then I realised that would mean a lot of changes in PSyclone. I suspect that I (we) should bite the bullet and make it all consistent. What do you think?

File: finite_difference/src/global_parameters_mod.f90 (hunk 2 above)
Comment (arporter) [reply finite_difference/src/global_parameters_mod.f90]: I think I probably could get rid of NAME_LEN. Although I see that I've declared ALIGNMENT as public, it really should be 'internal' to the library, i.e. application code shouldn't need it and therefore I don't think I need to re-name it.

File: finite_difference/src/kernel_mod.f90 (hunk 1 above)
Comment (hiker) [reply finite_difference/src/kernel_mod.f90]: *grin* Yes, I counted around 134 'pointwise' alone in the gocen1.0 Fortran test files :) Still, I would be strongly in favour of fixing it, and am happy to do the PSyclone fixes (if you can wait for a few days).

File: finite_difference/src/global_parameters_mod.f90 (hunk 2 above)
Comment (hiker) [reply finite_difference/src/global_parameters_mod.f90]: Agreed, and then make it private as well ;)

Comment (arporter) [reply finite_difference/src/global_parameters_mod.f90]: Done - I've moved it into `grid_mod` as that's the only place that uses it.

File: finite_difference/src/kernel_mod.f90 (hunk 1 above)
Comment (arporter) [reply finite_difference/src/kernel_mod.f90]: Yes, I agree (sigh). Everything is GO'd now...

**Comment by arporter** [comment]
Ready for a second review from @hiker if you'd be so kind :-)

**Review by hiker — APPROVED** [review]
Looks good now!

**Comment by hiker** [comment]
I hope the default merging option (create merge commit) was the right one.

Note I have created: https://github.com/stfc/PSyclone/issues/198 to update PSyclone.

**Pull request merged**